tests/activation/test_prompts.py

```diff
@@ -0,0 +1,280 @@
+"""test that prompt behavior is correct in supported shells"""
+from __future__ import absolute_import, unicode_literals
+
+import os
+import subprocess
+import sys
+from collections import namedtuple
+from textwrap import dedent
+
+import pytest
+
+import virtualenv
+
+try:
+    from pathlib import Path
+except ImportError:
+    from pathlib2 import Path
+
+# This must match the DEST_DIR provided in the ../conftest.py:clean_python fixture
+ENV_DEFAULT = "env"
+
+# This can be anything
+ENV_CUSTOM = "env_custom"
+
+# Standard prefix, surround the env name in parentheses and separate by a space
+PREFIX_DEFAULT = "({}) ".format(ENV_DEFAULT)
+
+# Arbitrary prefix for the environment that's provided a 'prompt' arg
+PREFIX_CUSTOM = "---ENV---"
+
+# Filename template: {shell}.script.(normal|suppress).(default|custom)[extension]
+SCRIPT_TEMPLATE = "{}.script.{}.{}{}"
+
+# Filename template: {shell}.out.(normal|suppress).(default|custom)
+OUTPUT_TEMPLATE = "{}.out.{}.{}"
+
+SHELL_LIST = ["bash", "fish", "csh", "xonsh", "cmd", "powershell"]
+
+
+# Py2 doesn't like unicode in the environment
+def env_compat(s):
```

Comment:
one character variables only within list/dict comprehension 👍

Reply:
Fixed in 5e5b5c1